### PR TITLE
R Resources blog post

### PR DESCRIPTION
**Pull request by ns-rse**
I had to provide some R resources to someone for a new project today. Recently a collaborator was also having to start out with learning R and I shared some of my links with them.

Figured it would be useful to make a blog post out of it so we've a resource to point people to in the future. It can of course be updated as new resources come online.

I've checked the majority of links should work (found a few dated ones for books that I corrected).

Feedback more than welcome, too much, too many sections, not enough focus?

@ubdbra001 you know more about R than me so if there is any glaring omission please let me know but if there are significant additional resources to be added I'd suggest making a separate issue/PR after this is published.

### CODE REVIEW

File: _posts/2023-10-06-r-resources.md [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+category:
+link:
+description:
+social_image: /assets/images/r_hex_collage.png
+type: text
+excerpt_separator: <!--more-->
+---
+
+
+[R](https://www.r-project.org) is a statistical programming languages and one of the most popular languages for data
```
Comment (Robadob) [inline _posts/2023-10-06-r-resources.md]: l
```suggestion
[R](https://www.r-project.org) is a statistical programming language and one of the most popular languages for data
```

File: _posts/2023-10-06-r-resources.md [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+analysis, statistics and plotting in academia and industry. Learning a new language can be daunting, particularly if you
+have no experience of scripting and are used to Graphical User Interfaces (GUIs) where you point and click to perform
+your statistical analysis.
+
+Fear not though, there are lot of resources and very friendly, enthusiastic and helpful R users out there who can help
+you on your journey learning [R](https://www.r-project.org). This post details some of them, and I'd welcome additions.
```
Comment (Robadob) [inline _posts/2023-10-06-r-resources.md]: > I'd welcome additions.

Bit of a useless call to action, if you're not saying how to contribute.

File: _posts/2023-10-06-r-resources.md [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+
+<!--more-->
+
+Most of these resources are links websites that are free and openly available. Where books are linked they are very
```
Comment (Robadob) [inline _posts/2023-10-06-r-resources.md]:
```suggestion
Most of these resources are links to websites that are free and openly available. Where books are linked they are very
```

File: _posts/2023-10-06-r-resources.md [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+
+R has its own Markdown language for writing literate documents and a comprehensive resources covering all aspects is [R
+Markdown: The Definitive Guide](https://bookdown.org/yihui/rmarkdown/) by Yihui Xie, J.J. Allaire and Garret
+Grolemund. By writing your work in R Markdown you are performing [literate
+programming](https://en.wikipedia.org/wiki/Literate_programming) and it means your report can updated automatically if
+the underlying data changes. Output to HTML, PDF, LibreOffice, Microsoft Office and many other formats. The underlying
+source can be version controlled using Git so that it is documented, backed up (e.g. on GitHub or GitLab) and it is easy
+to collaborate with colleagues.
+
+More recently [Posit (nee RStudio)](https://www.posit.com) have developed [Quarto](https://quarto.org) the next
```
Comment (Robadob) [inline _posts/2023-10-06-r-resources.md]:
```suggestion
More recently [Posit (new RStudio)](https://www.posit.com) have developed [Quarto](https://quarto.org) the next
```

File: _posts/2023-10-06-r-resources.md [hunk 5]
```diff
@@ -0,0 +1,301 @@
...
+### Git Book
+
+It is good practice to version control your code and literate documents as you develop them. This can be achieved using
+the version control system [Git](https://git.scm). Get yourself an account on [GitHub](https://github.com) and/or
+[GitLab](https://gitlab.com) and settle down to read Jenny Bryans excellent [Happy Git and GitHub for the
+useR](https://happygitwithr.com).
+
+### Tidyverse
+
+You will hear a lot about the [Tidyverse](https://www.tidyverse.org/) which _is an opinionated collection of R packages
```
Comment (Robadob) [inline _posts/2023-10-06-r-resources.md:114]:
```suggestion
You will hear a lot about the [Tidyverse](https://www.tidyverse.org/) which is an opinionated collection of R packages
```

File: _posts/2023-10-06-r-resources.md [hunk 6]
```diff
@@ -0,0 +1,301 @@
...
+Graphics](https://link.springer.com/book/10.1007/0-387-28695-0) has been very popular.
+
++ [ggplot2: Elegant Graphics for Data Analysis (3e)](https://ggplot2-book.org/index.html) by Hadley Wickham
++ [The R Graph Gallery – Help and inspiration for R charts](https://r-graph-gallery.com/index.html)
++ [Data visualization with R and ggplot2 | the R Graph Gallery](https://r-graph-gallery.com/ggplot2-package.html)
++ [ggplot2tor](https://ggplot2tor.com/)
+
+### Advanced Topics
+
+There is a lot to R than just Statistical analysis and one day you may want to investigate these in greater detail. The
```
Comment (Robadob) [inline _posts/2023-10-06-r-resources.md]:
```suggestion
There is a lot more to R than just Statistical analysis and one day you may want to investigate these in greater detail. The
```

File: _posts/2023-10-06-r-resources.md [hunk 7]
```diff
@@ -0,0 +1,301 @@
...
+## The R Journal
+
+The [R Journal](https://journal.r-project.org/) is the peer-reviewed, open-access scientific journal published by the R
+Foundation. It includes articles on packages, reviews and proposals, comparisons and benchmarking, applications of
+existing techniques and special issue articles to accompany conferences or particular topics.
+
+## Cheatsheats
+
+Cheatsheets come in handy as a reference to packages and commands. A central repository of cheatsheets is maintained by
+[Postit](https://posit.co/resources/cheatsheets/).
```
Comment (Robadob) [inline _posts/2023-10-06-r-resources.md]:
```suggestion
[Posit](https://posit.co/resources/cheatsheets/).
```

**Review by Robadob — COMMENTED** [review]
As you say a bit long and unweildy.

Possible tangential sections to purge:
* emacs
* git tools (and git kraken mention)
* mastodon
* generative art

Will leave it for Dan to approve, I've added various small comments for typographical errors I spotted whilst skimming.

File: _posts/2023-10-06-r-resources.md (hunk 2 above)
Comment (ns-rse) [reply _posts/2023-10-06-r-resources.md]: Good point, I'll stick a `mailto:` link in.

File: _posts/2023-10-06-r-resources.md (hunk 4 above)
Comment (ns-rse) [reply _posts/2023-10-06-r-resources.md]: Its meant to be [nee](https://duckduckgo.com/?q=define+nee&t=opera&ia=definition) as Posit used to be called RStudio but changed its name and so I'm using it in the same sense as a ladies maiden name.

Comment (Robadob) [reply _posts/2023-10-06-r-resources.md]: Not a term I'm familiar with so looks like a typo.

**Comment by ns-rse** [comment]
Thanks @Robadob good attention to detail as always.

* ESS/Emacs is one of the original ways of working with R in an IDE like environment as it goes back at least 20 years when I started using it so I thought it worth mentioning as an alternative to RStudio. Plus I've a personal rule to mention Emacs at least once in every blog post I ever write!
* I'm keen to encourage people who are just starting with R to version control their code so felt it important to mention Git, but in my opinion the amount of things you can do Git wise from RStudio is severely limited, hence the mention of GitKraken. Thought it made a nice segue to the Git course too.
* I included the Mastodon as `#rstats` was very popular on Tw@ter and now Mastodon and is a great way of finding out lots of neat and/or new things to do with R. Trying to encourage people to immerse themselves in the culture and become part of the community by giving them some sign-posts.
* Happy to strip the generative art, just thought it was something a bit different that might pique interest.

**Comment by Robadob** [comment]
Just perhaps a bit information overload for someone new to R. Perhaps better to highlight the best/most common options. E.g. if someone wants to know about R, saying USE GIT is just a distraction/irrelevant if they know nothing about version control/git.

Likewise, You may personally dislike Twitter. But it has 100-200x the active users of Mastodon. So chances are more readers will know how to use Twitter than Mastodon. Not that I think there's any value to advertising R influencers on their either. YouTube would probably be the better option, for people who prefer to consume learning materials in video form.

Each to their own though 🤷‍♂️

**Comment by ubdbra001** [comment]
Just had a look through and I think you've included pretty much everything that I'd put in.

On Git: I agree with Neil that it should be in there, but also agree with Rob that it's placed a bit to early on. It may be worth having an intermediate section (**Once you've got the basics** maybe) that highlights GIt resources as well as the Reproducible Analyitical Pipelines link as good practices that new R users should try and include as early as possible. Either that or having a second post that covers topics that are a bit more advanced/ancillary to just learning the language?

On Mastodon: I think highlighting that the #Rstats hashtag is a useful way of finding the R community there is a good inclusion.  It may also be worth saying that this is the same hashtag used on the place formerly known as twitter.
I do agree with Rob that the list of people and bots on Mastodon is a bit much (though I do agree these are really good people to follow), ~maybe make a list on Mastodon and link to it?~ Edit: Ignore that, Mastodon lists are not public

**Comment by ns-rse** [comment]
Thanks @ubdbra001 I have the `Advanced Topics` already in there which includes a link to _Reproducible Analytical Pipelines_ but have moved the whole Git section there.

I'm personally opposed to referencing Tw@ter in any way, shape or form regardless of its further reach (its self-reinforcing to not make any effort to change the _status quo_) but have referenced that it is the same hashtag used on that site.

**Pull request merged**